### PR TITLE
CDD-1650: incoming request headers logged

### PR DESCRIPTION
**Pull request by cichockimc**
as discussed here https://hmrcdigital.slack.com/archives/C0J8UQWDV/p1531486441000343

### CODE REVIEW

File: app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala [hunk 1]
```diff
@@ -45,6 +46,7 @@ class QueueController @Inject()(queueService: QueueService, fieldsService: ApiSu
   def save(): Action[AnyContent] = Action.async {
     implicit request => {
       val headers = request.headers
+      Logger.debug(s"[QueueController] Request received. Headers=$headers")
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala]: Do you need to log all headers? or just a specific one?
Let's NOT add noise if not needed.

Comment (AndySpaven) [reply app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala]: I think you also need to ensure these NEVER get logged in ET or PROD unless you can guarantee there is no PII in any of those headers.  Go GDPR ;-)

File: app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala [hunk 2]
```diff
@@ -45,6 +46,8 @@ class QueueController @Inject()(queueService: QueueService, fieldsService: ApiSu
   def save(): Action[AnyContent] = Action.async {
     implicit request => {
       val headers = request.headers
+      val message = headers.get("X-Conversation-ID").fold(s"[conversationId not found] Headers=$headers"){ cid => s"[conversationId=$cid]"}
+      Logger.debug(s"[QueueController]$message Request received")
```
Comment (gokyo) [inline app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala]: I would change it to:
```
Logger.debug(s"Request received - $message")
```

Comment (gokyo) [reply app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala]: If you want to include `[QueueController]` in the log message, you could also write `[QueueController#save()]`, so you know what method has been called

Comment (cichockimc) [reply app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala]: @gokyo changed and pushed.

**Pull request merged**